packages/twenty-server/src/utils/apply-cors-to-exceptions.ts

```diff
@@ -7,24 +7,27 @@ import { Response } from 'express';
 // This class add CORS headers to exception response to avoid misleading CORS error
 @Catch()
 export class ApplyCorsToExceptions implements ExceptionFilter {
+  private readonly logger = new Logger(ApplyCorsToExceptions.name);
   catch(exception: any, host: ArgumentsHost) {
-    const ctx = host.switchToHttp();
-    const response = ctx.getResponse<Response>();
+    try {
+      const ctx = host.switchToHttp();
+      const response = ctx.getResponse<Response>();
 
-    if (!response.header) {
-      return;
-    }
+      response.header('Access-Control-Allow-Origin', '*');
+      response.header(
+        'Access-Control-Allow-Methods',
+        'GET,HEAD,PUT,PATCH,POST,DELETE',
+      );
+      response.header(
+        'Access-Control-Allow-Headers',
+        'Origin, X-Requested-With, Content-Type, Accept',
+      );
 
-    response.header('Access-Control-Allow-Origin', '*');
-    response.header(
-      'Access-Control-Allow-Methods',
-      'GET,HEAD,PUT,PATCH,POST,DELETE',
-    );
-    response.header(
-      'Access-Control-Allow-Headers',
-      'Origin, X-Requested-With, Content-Type, Accept',
-    );
+      response.status(exception.getStatus()).json(exception.response);
```

Comment:
if the issue is with exception getStatus, I would check if exception is defined rather than putting a try catch